grin/src/Grin/ExtendedSyntax/TypeEnv.hs

```diff
@@ -0,0 +1,187 @@
+{-# LANGUAGE LambdaCase, RecordWildCards #-}
+module Grin.ExtendedSyntax.TypeEnv
```

Comment:
This module seems to be unnecessarily duplicated. What do I miss?

Reply:
I guess its due to the imported modules (dependencies)

Reply:
My approach is to duplicate everything. Then the transition will be straightforward when everything is finished. Also, if we have shared modules, there might instance clashes which are hard to investigate (encountered those recently).

Reply:
Maybe this will give us some headache in the long run :)